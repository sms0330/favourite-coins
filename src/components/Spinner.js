import React from 'react';

export const Spinner = props => {
  return (
    <div className="ui segment" style={{ minHeight: "10em" }}>
      <div className="ui active inverted dimmer">
        <div className="ui text loader">{props.message} Loading</div>
      </div>
      <p></p>
    </div>
  );
};

export default Spinner;