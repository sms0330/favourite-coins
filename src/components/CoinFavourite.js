import React, { Component } from 'react';
import Bitcoin from './assets/img/Bitcoin.svg';
import BitcoinCash from './assets/img/BCH.svg';
import Ethereum from './assets/img/Ethereum.svg';
import Litecoin from './assets/img/Litecoin-new.svg';
import XRP from './assets/img/XRP.svg';
import Spinner from './Spinner';

const images = {
  BITCOIN: Bitcoin,
  'BITCOIN CASH': BitcoinCash,
  ETHEREUM: Ethereum,
  XRP: XRP,
  LITECOIN: Litecoin,
};
class CoinFavourite extends Component {
  constructor(props) {
    super(props);
    this.state = {
      loading: false,
      favourites: [],
    };
  }

  deleteFavourite(id) {
    this.setState(state => {
      return {
        favourites: state.favourites.filter(f => f.id !== id),
      };
    });
  }

  render() {
    if (!this.state.favourites) {
      return <Spinner />;
    }
    return (
      <div className="head">
        <h1 className="head title">Favourite Coins</h1>
        {this.props.favourites.map((el, i) => {
          return (
            <button key={i} onClick={() => this.props.remove(i)} className="faveButtons">
              <span id="favorite">
                <span style={{ verticalAlign: 'middle' }}>
                  <img src={images[el]} alt="logo" />
                </span>
                <span style={{ verticalAlign: 'middle' }}> {el}</span>
              </span>
            </button>
          );
        })}
      </div>
    );
  }
}

export default CoinFavourite;
