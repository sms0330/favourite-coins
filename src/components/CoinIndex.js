import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { Coin } from '../requests';
import Spinner from './Spinner';

class CoinIndex extends Component {
  constructor(props) {
    super(props);
    this.state = {
      coins: [],
    };
  }
  componentDidMount() {
    Coin.index().then(coins => {
      const filteredCoins = coins.filter(c => {
        if (
          c.id === 'bitcoin' ||
          c.id === 'ethereum' ||
          c.id === 'ripple' ||
          c.id === 'bitcoin-cash' ||
          c.id === 'litecoin'
        ) {
          return true;
        }
        return false;
      });
      this.setState({
        coins: filteredCoins,
      });
    });
  }

  render() {
    if (!this.state.coins) {
      return <Spinner />;
    }
    return (
      <main>
        <h1>ALL COINS</h1>
        <table className="ui fixed table">
          <thead>
            <tr>
              <th>NAME</th>
              <th>TOTAL SUPPLY</th>
              <th>MARKET CAP</th>
              <th>CURRENT PRICE</th>
            </tr>
          </thead>

          {/*  */}
          <tbody>
            {this.state.coins.map((coin, index) => (
              <tr key={index}>
                <td data-label="NAME">
                  <Link to={`/coins/${coin.id}`}>{coin.name.toUpperCase()}</Link>
                </td>
                <td data-label="TOTAL SUPPLY">{coin.total_supply || 'INSERT VALUE'}</td>
                <td data-label="MARKET CAP">{coin.market_cap || 'INSERT VALUE'}</td>
                <td data-label="CURRENT PRICE">${coin.current_price || 'INSERT VALUE'}</td>
              </tr>
            ))}
          </tbody>
        </table>
        <br/>
      </main>
    );
  }
}

export default CoinIndex;