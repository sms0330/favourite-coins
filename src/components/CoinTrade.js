import React, { Component } from 'react';
import { Coin } from '../requests';
import Spinner from './Spinner';

class CoinTrade extends Component {
  constructor(props) {
    super(props);
    this.state = {
      coins: [],
      target: null,
      targetPrice: null,
      targetAmount: null,
      buy: null,
      sell: null,
      receipt: null,
      value: null,
    };
  }

  componentDidMount() {
    Coin.index().then(coins => {
      const filteredCoins = coins.filter(c => {
        if (
          c.id === 'bitcoin' ||
          c.id === 'ethereum' ||
          c.id === 'ripple' ||
          c.id === 'bitcoin-cash' ||
          c.id === 'litecoin'
        ) {
          return true;
        }
        return false;
      });
      this.setState({
        coins: filteredCoins,
      });
    });
  }

  GetDropdownCoinPrice = e => {
    e.preventDefault();
    const coin = this.state.coins.find(coin => coin.id === e.target.value);
    const price = coin.current_price;
    this.setState({ dropdownCoinPrice: price });
  };

  GetAmount = e => {
    e.preventDefault();
    const amount = e.target.value;
    this.setState({ targetAmount: amount });
  };

  Getprice(e) {
    e.preventDefault();
    const value = e.target.value;
    const coin = this.state.coins.find(coin => {
      return coin.name === value;
    });
    const price = coin.current_price;
    this.setState({ target: coin.symbol });
    this.setState({ targetPrice: price });
  }

  Buy = e => {
    e.preventDefault();
    this.setState({ buy: true });
    this.setState({ sell: false });
  };

  Sell = e => {
    e.preventDefault();
    this.setState({ buy: false });
    this.setState({ sell: true });
  };

  Submit = e => {
    const { target, targetAmount, targetPrice, buy, sell } = this.state;
    e.preventDefault();
    const refPrice = this.props.info.market_data.current_price.cad;
    const refSymbol = this.props.info.symbol;
    console.log('targetPrice is ' + targetPrice);
    console.log('targetAmount is ' + targetAmount);
    const coinNumToRef = refPrice / (targetPrice * targetAmount);
    const coinNumToTarget = (targetPrice * targetAmount) / refPrice;
    if (buy === true && sell === false){
      this.setState({
        receipt: `You have parchased ${
          targetAmount
        } ${target ? target.toUpperCase() : null} for ${coinNumToTarget} ${refSymbol.toUpperCase()}`,
      });
    } else if (buy === false && sell === true){
      this.setState({
        receipt: `You have sold ${
          targetAmount
        } ${target ? target.toUpperCase() : null} for ${coinNumToRef} ${refSymbol.toUpperCase()}`,
      });
    } 
  };

  render() {
    if (!this.state.coins) {
      return <Spinner />;
    }
    return (
      <form>
        <div className="two column row">
          <div className="trade">
            <br />
            <button className="button" onClick={this.Buy}>
              BUY
            </button>
            <button className="button" onClick={this.Sell}>
              SELL
            </button>
          </div>
          <div />
          <br />
          <div className="dropdown currency">
            <select
              onChange={e => {
                this.Getprice(e);
              }}
              className="ui fluid search dropdown"
            >
              <option selected disabled>
                Select Trade Currency{' '}
              </option>
              {this.state.coins.map(coin => {
                if (coin.name !== this.props.info.name) {
                  return (
                    <option key={coin.name} value={coin.name}>
                      {coin.name}
                    </option>
                  );
                } else { return false }
              })}
            </select>
          </div>
          <br />
          <div className="ui form">
            <input
              type="number"
              min="0"
              placeholder="Amount"
              value={this.state.amount}
              onChange={this.GetAmount}
            />
            <br />
          </div>
          <br />
          <div className="formButton">
            <button
              className="button"
              onChange={this.GetPrice}
              onClick={e => {
                this.Submit(e);
              }}
            >
              <h5>SUBMIT</h5>
            </button>
          </div>
        </div>
        <div className="receipt">{this.state.receipt}</div>
      </form>
    );
  }
}

export default CoinTrade;
