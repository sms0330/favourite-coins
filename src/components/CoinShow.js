import React, { Component } from 'react';
import { Coin } from '../requests';
import CoinTrade from './CoinTrade';
import Spinner from './Spinner';

class CoinShow extends Component {
  constructor(props) {
    super(props);
    this.state = { coin: null, showingCoin: null };
  }
  
  componentDidMount() {
    Coin.show(this.props.match.params.id).then(coin => {
      this.setState({
        coin: coin
      });
    });
  }

  componentDidUpdate(prevProps, prevState) {
    if (this.props.match.params.id !== prevProps.match.params.id) {
      Coin.show(this.props.match.params.id).then(coin => {
        this.setState({
          coin: coin
        });
      });
    }
  }

  render() {
    if (!this.state.coin) {
      return <Spinner />;
    }
    const { name, symbol, image, market_data } = this.state.coin;
    return (
      <main>
        <div className="ui segment">
          <div className="ui two column grid">
            <div className="column" id="show-column">
              <div className="ui form">
                <div className="field">
                  <label>
                    <img src={image.thumb} style={{verticalAlign:"middle"}} className='logo' alt='logo'/>
                    <span style={{verticalAlign:"middle"}}>{name.toUpperCase()}</span>
                    <span style={{verticalAlign:"middle"}}>{`(${symbol.toUpperCase()})`}</span>
                  </label>
                </div>
                <hr className="show-line" />
                <p><strong>CURRENT PRICE:</strong> ${market_data.current_price.cad}</p>
                <p><strong>MARKET CAP:</strong> {market_data.market_cap.cad}</p>
                <p><strong>LOW 24H:</strong> {market_data.low_24h.cad}</p>
                <p><strong>HIGH 24H:</strong> {market_data.high_24h.cad}</p>
                <p><strong>CIRCULATING SUPPLY:</strong> {market_data.circulating_supply}</p>
                <p><strong>TOTAL SUPPLY:</strong> {market_data.total_supply}</p>
                <p><strong>MARKET CAP RANK:</strong> {market_data.market_cap_rank}</p>
                <div className="favourite">
                  <small onClick={(e) => this.props.passToParent(e, name.toUpperCase())}> ‚®Å ADD TO FAVOURITE</small>
                </div>
              </div>
            </div>
            <div className="column">
              <CoinTrade info={this.state.coin} />
            </div>
          </div>
        </div>
      </main>
    );
  }
}

export default CoinShow;