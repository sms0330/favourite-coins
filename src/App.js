import React, {Component} from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import CoinIndex from './components/CoinIndex';
import CoinFavourite from './components/CoinFavourite';
import Spinner from "./components/Spinner";
import CoinShow from './components/CoinShow';

const saveData = JSON.parse(localStorage.getItem("favourites"));
class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      loading: false, 
      favourites: saveData ? saveData : [],
    }
  }

  componentDidMount() {
    const savedFavourites = JSON.parse(localStorage.getItem("favourites"));
    if (savedFavourites !== null) {
      this.setState(savedFavourites)
    }
    console.log('Component did mount')
  }

  componentDidUpdate(prevProps, prevState) {
    if (prevState.favourites !== this.state.favourites) {
      localStorage.setItem('favourites', JSON.stringify(this.state.favourites))
        // this.setState( )
        console.log('changed')
    }
  }

  childCallback(e, value) {
    // value passed from child
    if (!this.state.favourites.includes(value)) {
    e.preventDefault();
    console.log(value);
    console.log(this.state)
    this.setState({favourites: [...this.state.favourites, value]});
    }
  }

  remove(i) {
    this.state.favourites.splice(i,1);
    this.setState({favourites: [...this.state.favourites]})
    console.log(this.state.favourites)

  }

  render() {
    if (this.state.loading) {
      return <Spinner />;
    }
    return (
      <div className="App">
        <Router>
          <CoinFavourite favourites={this.state.favourites} remove={this.remove.bind(this)}/>
          <br/>
          <CoinIndex />
            <Switch>
              <Route
                exact
                path="/coins/:id"
                render={(props) => <CoinShow {...props} passToParent={this.childCallback.bind(this)} />}

              />
            </Switch>
        </Router>
      </div>
    );
  }
}

export default App;