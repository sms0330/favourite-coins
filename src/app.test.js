import { render, screen } from '@testing-library/react';
import App from './App';
import Hello from './components/Test';
import CoinIndex from './components/CoinIndex';
import CoinShow from './components/CoinShow';
import CoinFavourite from './components/CoinFavourite';
import CoinTrade from './components/CoinTrade';
import Spinner from "./components/Spinner";

const user = {
    // name: "mike",
    name: "tom",
    age: 30,
};

const user2 = {
    age:20,
}

test('Check to include hello', () => {
    render(<Hello user={user}/>);
    const Element = screen.getByText(/Hello/i);
    expect(Element).toBeInTheDocument();
})

test('snapshot : name vaild', () => {
    const Element = render(<Hello user={user}/>);
    expect(Element).toMatchSnapshot();
});

test('snapshot : name invaild', () => {
    const Element = render(<Hello user={user2}/>);
    expect(Element).toMatchSnapshot();
});

test('snapshot : react app', () => {
    const linkElement = render(<App/>);
    expect(linkElement).toMatchSnapshot();
});
  
test('snapshot : CoinIndex', () => {
  const linkElement = render(<CoinIndex/>);
  expect(linkElement).toMatchSnapshot();
});
  
test('snapshot : CoinShow', () => {
  const linkElement = render(<CoinShow/>);
  expect(linkElement).toMatchSnapshot();
});

test('snapshot : CoinFavourite', () => {
    const linkElement = render(<CoinFavourite/>);
    expect(linkElement).toMatchSnapshot();
});

test('snapshot : CoinTrade', () => {
    const linkElement = render(<CoinTrade/>);
    expect(linkElement).toMatchSnapshot();
});

test('snapshot : Spinner', () => {
    const linkElement = render(<Spinner/>);
    expect(linkElement).toMatchSnapshot();
});