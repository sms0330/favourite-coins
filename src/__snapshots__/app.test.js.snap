// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot : CoinIndex 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ui segment"
        style="min-height: 10em;"
      >
        <div
          class="ui active inverted dimmer"
        >
          <div
            class="ui text loader"
          >
             Loading
          </div>
        </div>
        <p />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ui segment"
      style="min-height: 10em;"
    >
      <div
        class="ui active inverted dimmer"
      >
        <div
          class="ui text loader"
        >
           Loading
        </div>
      </div>
      <p />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`snapshot : CoinTrade 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ui segment"
        style="min-height: 10em;"
      >
        <div
          class="ui active inverted dimmer"
        >
          <div
            class="ui text loader"
          >
             Loading
          </div>
        </div>
        <p />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ui segment"
      style="min-height: 10em;"
    >
      <div
        class="ui active inverted dimmer"
      >
        <div
          class="ui text loader"
        >
           Loading
        </div>
      </div>
      <p />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`snapshot : Spinner 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ui segment"
        style="min-height: 10em;"
      >
        <div
          class="ui active inverted dimmer"
        >
          <div
            class="ui text loader"
          >
             Loading
          </div>
        </div>
        <p />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ui segment"
      style="min-height: 10em;"
    >
      <div
        class="ui active inverted dimmer"
      >
        <div
          class="ui text loader"
        >
           Loading
        </div>
      </div>
      <p />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`snapshot : name invaild 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button>
        Login
      </button>
    </div>
  </body>,
  "container": <div>
    <button>
      Login
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`snapshot : name vaild 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1>
        Hello! 
        tom
      </h1>
    </div>
  </body>,
  "container": <div>
    <h1>
      Hello! 
      tom
    </h1>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`snapshot : react app 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div
          class="head"
        >
          <h1
            class="head title"
          >
            Favourite Coins
          </h1>
        </div>
        <br />
        <main>
          <h1>
            ALL COINS
          </h1>
          <table
            class="ui fixed table"
          >
            <thead>
              <tr>
                <th>
                  NAME
                </th>
                <th>
                  TOTAL SUPPLY
                </th>
                <th>
                  MARKET CAP
                </th>
                <th>
                  CURRENT PRICE
                </th>
              </tr>
            </thead>
            <tbody />
          </table>
          <br />
        </main>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div
        class="head"
      >
        <h1
          class="head title"
        >
          Favourite Coins
        </h1>
      </div>
      <br />
      <main>
        <h1>
          ALL COINS
        </h1>
        <table
          class="ui fixed table"
        >
          <thead>
            <tr>
              <th>
                NAME
              </th>
              <th>
                TOTAL SUPPLY
              </th>
              <th>
                MARKET CAP
              </th>
              <th>
                CURRENT PRICE
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
        <br />
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
